account_financial_report_webkit/models/account_static_balance.py

```diff
@@ -18,31 +18,42 @@ class AccountStaticBalance(models.Model):
         comodel_name='account.account', required=True, index=True)
     period_id = fields.Many2one(
         comodel_name='account.period', required=True, index=True)
+    journal_id = fields.Many2one(
+        'account.journal', required=True, index=True)
     credit = fields.Float(default=0.0)
     debit = fields.Float(default=0.0)
     balance = fields.Float(default=0.0)
     curr_balance = fields.Float(default=0.0)
 
     _sql_constraints = [
         ('account_static_balance_unique',
-         'unique (account_id, period_id)',
-         'Only 1 unique combination of account and period is allowed.'),
+         'unique (account_id, period_id, journal_id)',
+         'Only 1 unique combination of account, journal and period is '
+         'allowed.'),
     ]
 
     @api.multi
     def get_entries_to_calculate(self, periods=None):
         """ Calculates all possible combinations of accounts and closed
-        periods which are missing from the expected result.
+        periods which are missing from the expected result. Also clear
+        account_static_balance from integrity issue records.
         """
+        self._validate_integrity()
```

Comment:
Isn't this method called both when static balances are generated (which is when we want to clean up outdated balances) as well as when balances are used for report generation (which is when we don't want to make database mutations)?

Reply:
BTW we still want to exclude the entries in the query where we fetch the balances for report generation.

Reply:
The report generation happens in the function `model` which contains a fallback to calculate the missing balances on the fly, but doesn't save them. So we need to delete them in the `get_entries_to_calculate` so that it will be recalculated and shown correctly in the report. Or am i missing something?

Reply:
Yes, if we could always delete them here that would work to ensure the data is up to date but unfortunately we cannot always delete them here because of the concurrency model in Postgresql.

Reply:
Added the unlink here: https://github.com/EssentNovaTeam/account-financial-reporting/pull/6/commits/700b90476b46e3a6f2ce7234c493130bbd0bca42#diff-c4749f6b2ebab521a3de0b3add26b25c49bc78e8c8a4271f430e91f2cf6b4f54R131-R135